### ISSUE TITLE
Emails in Person History Get Attributed To Wrong Sender Under Certain Conditions

### ISSUE THREAD

**Issue by JimMichael** [SparkDevNetwork/Rock#5456]
## Description
Under certain conditions -- specifically, an email sent via a Workflow _Send Email_ action with _Save Communication History_ enabled -- the wrong "sender" can be recorded in the recipient person's history IF the sender's email address resides on more than one Person in Rock.

## Steps to Reproduce

1. Pick two test people and give them the same email address. Person 1 will be the "Sender" and should have a higher PersonId than the other person (Person 2) that has the same email. For example, an Adult person could have their email on their record, but their child (with a lower PersonId) also has the same email on her record.
1. Create a workflow that has two Person attributes such as _Person_ (the email recipient) and _Sender._
1. Edit the Sender attribute and set its default to the _Person 1_ you identified in step 1 with the higher PersonId.
1. Edit the Person attribute and set its default to a third person (neither of the two people you identified in Step 1, but one you can receive the email to).
1. Configure an Email Send action to send something _to_ the Person attribute, _from_ the Sender attribute, with _Save Communication History_ enabled.
1. Trigger the workflow and the email should be sent. Now look in the recipient's History and the email should be there... HOWEVER, it will show as from the _wrong person_ . It should show as from Person 1, but actually shows as from Person 2... presumably because Rock looked up the first person it could find with that email address and hit the one with the lower PersonId. This seems to be happening somewhere outside of the workflow and during the process that writes to history.

Here's a completed workflow showing that the recipient (Person) and Sender are correct.

![image](https://github.com/SparkDevNetwork/Rock/assets/3903408/379152b6-35b6-45e6-95a0-05b866cd20e0)And here's what shows in Karent Test's History (should be Karen Rossi, not Sxxxx Rossi, the person with Karen's email address on their record)

![image](https://github.com/SparkDevNetwork/Rock/assets/3903408/59c48222-ab85-41c2-b116-040ae8a97aa1)

**Expected behavior:**
Person History would show the correct sender on communications even if the sender's email address is on one or more other Persons.

**Actual behavior:**
History can show the wrong person as the sender, leading to much confusion. It seems like Rock is doing a simple "who has this email address?" lookup when determining a sender for Person History communications instead of looking up the email address of the actual person that sent the communication.

## Versions

* **Rock Version:** 14.2, 15.0

**Comment by dataCollegechurch** [comment]
Hi Jim,

I can confirm that this issue does indeed exist, but I am not sure I would classify it as a bug. The code appears to be working as it was designed. This probably needs to be entered as an enhancement request. Another scenario where something like this could occur is while using the communication wizard. In that scenario a Rock Person is not selected, instead a From Person Text Value and From Email Address is set.

**Comment by nairdo** [comment]
Hi @JimMichael I spent some time digging into this one too.  Tony is right about this and it is the current design of the underlying code.  There is a `FromName` and a `FromEmail` that the workflow action can set, but our RockEmailMessage code does not have a way to set a from `Person` today.  

I don't want to suggest that your idea is _not_ a good one, it would take deeper engineering and time/effort (vs a quick bug fix).  So, I'd like to suggest moving this to the [Ideas board](https://community.rockrms.com/Ideas) if you still think it's worth pursuing.

**Comment by JimMichael** [comment]
I disagree with this assessment, but will create an enhancement for it. I disagree because Rock is literally recording wrong/incorrect data when saving the "sender" to Person History. As I stated recently in another issue, I totally "get" the "this code is working as designed" argument and it's an easy-out to allow an issue to be closed without effort, but sometimes the _design is just wrong._ 

Please don't add any perceived anger or malice to my statement. I'm just trying to convey a thought position, nothing more.

**Comment by zackdutra** [comment]
Hey team, just wanted to voice that we also strongly disagree about this being an enhancement.  This can lead to false information about who sent an email, creating misinformation and confusion. This is a bug, and @JimMichael  was right to bring it up. We appreciate the effort the team puts into classifying issues, but a communication history log needs to accurately reflect who sent an email. If we can't accurately list the name of the person who sent it, perhaps we need to dumb it down and just say "from example@example.com" and not list a name - at least that would be an accurate statement. 

We would like to encourage @nairdo to reconsider. Similar to Jim, not mad here but do want to voice that we don't feel it's a small thing to dismiss fixing misinformation bugs as an enhancements.

**Comment by nairdo** [comment]
@zackdutra & @JimMichael  We were discussing this yesterday too and we've decided we will tackle this as an issue.  So please let us get this into the queue and we'll let you know when it's ready for release.

**Comment by JimMichael** [comment]
Thank you very much @nairdo!

**Comment by zackdutra** [comment]
Much appreciated @nairdo !
